### ISSUE TITLE
Ability to  'Hide from publishing'  a single token

### ISSUE THREAD

**Issue by keeganedwin** [tokens-studio/figma-plugin#2193]
**Is your feature request related to a problem? Please describe.**
In Figma, there is a feature that allows users to hide variables from publishing. However, when syncing and creating variables within Tokens Studio, the 'Hide from publishing' checkbox gets reset.

**Describe the solution you'd like**
A feature enhancement in Tokens Studio that allows for preserving the 'Hide from publishing' setting. This enhancement would greatly reduce the number of component variables that do not need to be published or visible to other designers.

------

This is somewhat tied to the work in #2194 - and could possibly be done in parallel,

```[tasklist]
### Related
- [ ] https://github.com/tokens-studio/figma-plugin/issues/2194
- [ ] https://github.com/tokens-studio/figma-plugin/issues/2186
```

**Comment by LukeFinch** [comment]
Is this a duplicate of #2186 ? Or is intended behaviour for if a user imports/syncs variables into the plugin?

**Comment by six7** [comment]
this one is for the individual token, the other one for whole theme groups (collections)

**Comment by LukeFinch** [comment]
Worth considering with this:
Whilst it's possible to hide a **style** from publishing - it's not possible to do this through the API (has to be done manually by a user in the assets panel). 

This and #2186 are very much **variable** specific. And should likely be directly tied in with the work for #2194

**Comment by six7** [comment]
> Whilst it's possible to hide a style from publishing - it's not possible to do this through the API (has to be done manually by a user in the assets panel).

technically a user could name a style beginning with a `_` so it's hidden - but it's a bit of a hacky workaround i feel.

agree that the work around this should just be tied to Variables 👍

**Comment by esthercheran** [comment]
Closing this ticket as this is covered in #2300
